language/move-lang/src/parser/syntax.rs

```diff
@@ -383,11 +386,38 @@ fn parse_value<'input>(tokens: &mut Lexer<'input>) -> Result<Value, Error> {
             tokens.advance()?;
             Value_::Bool(false)
         }
+        Tok::U8Value => {
+            let mut s = tokens.content();
+            if s.ends_with("u8") {
+                s = &s[..s.len() - 2]
+            }
+            let i = u8::from_str(s).unwrap();
+            tokens.advance()?;
+            // FIXME
+            Value_::U64(i.into())
+            // Value_::U8(i)
+        }
         Tok::U64Value => {
-            let i = u64::from_str(tokens.content()).unwrap();
+            let mut s = tokens.content();
+            if s.ends_with("u64") {
+                s = &s[..s.len() - 3]
+            }
+            let i = u64::from_str(s).unwrap();
             tokens.advance()?;
             Value_::U64(i)
         }
+        Tok::U128Value => {
+            let mut s = tokens.content();
+            if s.ends_with("u128") {
+                s = &s[..s.len() - 4]
+            }
+            let i = u128::from_str(s).unwrap();
+            tokens.advance()?;
+            // FIXME
+            // Note: Remove the use of std::convert::TryInto when this is resolved.
+            Value_::U64(i.try_into().unwrap())
+            // Value_::U128(i)
```

Comment:
I assume we'll fix this in the compiler at a later time.

Reply:
Yes, I checked with Todd and I will hold onto this for now and update it after we land https://github.com/libra/libra/pull/2594